### PR TITLE
added destination to case.results.download() functionality

### PR DESCRIPTION
**Pull request by maciej-flexcompute**
https://flow360.atlassian.net/browse/FLPY-3

- renamed download_manager() to download()
- added ‘destination’ parameter
- removed download_file() and other download methods
- renamed Flow360Resource.download_file() to Flow360Resource._download_file() to be private
- allow for providing to_file - which will save to file and add file extension when needed, or to_folder

### CODE REVIEW

File: flow360/component/case.py [hunk 1]
```diff
@@ -447,18 +447,6 @@ def results(self) -> CaseResults:
         """
         return self._results
 
-    def download_log(self, log_file, to_file=".", keep_folder: bool = True):
```
Comment (jinlin-flex) [inline flow360/component/case.py:450]: will need to refactor test cases that call this function in _test_surface_mesh.py and _test_volume_mesh.py

**Review by jinlin-flex — CHANGES_REQUESTED** [review]

File: examples/case_results.py [hunk 2]
```diff
@@ -27,15 +27,14 @@
 print(case.results.minmax_state.raw)
 
 # download all result files of the case:
-case.results.download_manager(all=True)
+downloaded_files = case.results.download(all=True, destination=case.name)
+print(downloaded_files)
```
[the remark this answers is not in the document]
Comment (maciej-flexcompute) [reply examples/case_results.py:31]: this is part of examples. I think it is better to use `print` in examples as these will be run and modify by users so they will not be familiar with log.

File: examples/case_results.py [hunk 3]
```diff
@@ -27,15 +27,14 @@
...
+
 
 # download specific result files of the case:
-case.results.download_manager(bet_forces=True, actuator_disk_output=True)
+case.results.download(bet_forces=True, actuator_disk_output=True)
 
 # alternative way of downloading using dedicated functions:
```
[the remark this answers is not in the document]
Comment (maciej-flexcompute) [reply examples/case_results.py]: removed

File: flow360/cloud/s3_utils.py [hunk 4]
```diff
@@ -52,7 +53,9 @@ def __call__(self, bytes_chunk_transferred):
 )
 
 
-def create_base_folder(path: str, target_name: str, to_file: str = ".", keep_folder: bool = True):
+def create_base_folder(
```
[the remark this answers is not in the document]
Comment (maciej-flexcompute) [reply flow360/cloud/s3_utils.py:56]: updated

File: flow360/component/case.py (hunk 1 above)
Comment (maciej-flexcompute) [reply flow360/component/case.py:450]: removed files _test_surface_mesh.py and _test_volume_mesh.py as we should not be running tests that require cloud access

File: flow360/cloud/s3_utils.py [hunk 5]
```diff
@@ -24,6 +24,7 @@
 
 from ..environment import Env
 from ..exceptions import CloudFileError
```
Comment (jinlin-flex) [inline flow360/cloud/s3_utils.py:26]: it seems like CloudFileError is unused

**Review by jinlin-flex — APPROVED** [review]

**Pull request merged**